### PR TITLE
NLogTraceListener - Reduce code complexity of ProcessLogEventInfo

### PR DESCRIPTION
**Pull request by snakefoot**

### CODE REVIEW

**Comment by 304NotModified** [comment]
Hi, is this refactored by a tool, or do I need to double-check it? (E.g. diff)

**Comment by snakefoot** [comment]
It was done with tool, but with a small adjustment afterwards to avoid returning updated loggerName


Sent from my Sony Xperia


---- Julian Verdurmen wrote ----


Hi, is this refactored by a tool, or do I need to double-check it? (E.g. diff)

File: src/NLog/NLogTraceListener.cs [hunk 1]
```diff
@@ -457,6 +430,37 @@ protected virtual void ProcessLogEventInfo(LogLevel logLevel, string loggerName,
...
 
+        private ILogger GetLogger(string loggerName, out StackTrace stackTrace, out int userFrameIndex)
+        {
+            loggerName = (loggerName ?? Name) ?? string.Empty;
+
+            stackTrace = null;
+            userFrameIndex = -1;
+            if (AutoLoggerName)
+            {
+                stackTrace = new StackTrace();
```
Comment (304NotModified) [inline src/NLog/NLogTraceListener.cs]: Double check, it isn't a problem that `new StackTrace` has been moved to an inner method?

Comment (snakefoot) [reply src/NLog/NLogTraceListener.cs]: Should not have any issues. But just to be sure, then I have moved capture of StackTrace back to original location.

**Comment by 304NotModified** [comment]
travis seems to be down for now

**Comment by snakefoot** [comment]
> travis seems to be down for now

No your `dependabot[bot]` has just made a huge queue of meaningless builds, which it is spinning up and down build-server to handle.

**Comment by 304NotModified** [comment]
ah thanks!

**Comment by 304NotModified** [comment]
travis should not build closed prs. That should be fixed now

**Comment by snakefoot** [comment]
@304NotModified Looks like your "fix" have now stopped travis from building anything :)

**Comment by snakefoot** [comment]
@304NotModified But it seems to build nice for NLog.Extension.Logging: https://travis-ci.org/NLog

**Comment by snakefoot** [comment]
@304NotModified Now it is building again: https://travis-ci.org/NLog/NLog/builds/512191462

**Comment by 304NotModified** [comment]
Build successfully, merged, done! Thanks!

**Pull request merged**